assignments/week_2/exercise_2_3/exercise_2_3.md

```diff
@@ -0,0 +1,360 @@
+of numbers. Likely you have heard of binary and hexidecimal. Our goal is to write
+a class that will allow you to convert between the different types of numbers.
+
+# Background
+There are a lot of different representations of numbers. The standard way we represent
+numbers are in base 10, the digits 0-9 are used in sequence to represent larger numbers.
+You could write out a large number like 25 as the following
+
+```
+10^0 * 5 + 10^1*2 = 5 + 20 = 25
+```
+
+We will be using the same approach but with different base numbers in this exercise.
+
+## Binary
+
+Binary is when you represent the number as only {1,0} this is a base two representation.
+If we take the number (the preceding b lets us know it is in binary)
+
+```
+b11001
+```
+
+that can bet written out as
+
+```
+1 * 2^0 + 0*2^1 + 0*2^2 + 1*2^3 + 1*2^4 = 1 + 0 + 0 + 8 + 16 = 25
+```
+
+## Hex
+
+Hex has a representation of numbers in base 16, now we only have 10 digits to represent numbers
+with, so we will need to include some letters to represent the actual number.
+Here we use the letters {a,b,c,d,e,f} to represent {10,11,12,13,14,15}.
+All hex numbers are typically prefaced with **0x** like below1
+
+```
+0x19 = 9 * 16^0 + 1*16^1 = 9 + 16 = 25
+```
+
+# Implementation
+
+Today we will be writing a class from scratch in order to convert between the
+different formats of the numbers. Furthermore, we will be keeping track of
+how many conversions we do between each type inside our class.
+
+## Starter Code
+There are two methods that have been implemented for you, they are below.
+They use the actual ASCII numbers of the different characters to convert
+hex to digits and digits to hex characters. Don't worry about how this works.
+Here we name our class NumberConverter, you can name it whatever you want.
+
+```c++
+/**
+ * returns a number 0-15 based off of a hex char
+ * This uses ascii to figure out number it corresponds to
+ * @param input
+ * @return
+ */
+int NumberConvereter::hexCharToNumber(char input) {
+  // this means it is [A,B,C,D,E,F]
+  if(input > 57) {
+    return input - (65 - 10);
+  } else {
+    // this is a digit in ascii
+    return input - 48;
+  }
+}
+
+/**
+ * returns the character that is the given hex number
+ * only valid on 0-15
+ * @param input
+ * @return
+ */
+char NumberConvereter::numberToHexChar(int input) {
+  if(input < 10) {
+    // it is a digit
+    return input + 48;
+  } else {
+    return (input - 10) + 65;
+  }
+}
+```
+
+There are multiple online resources to convert between the three types of numbers.
+I would just hard code an example
+
+## Part 0 Creating a class
+
+create a .h and .cpp called whatever you would like, then you will need to add the .cpp
+to the cmake file like so (ours is named NumberConverter.cpp)
+
+```cmake
+add_executable(exercise_2_3 exercise_2_3.cpp NumberConverter.cpp)
+```
+
+to build and run your code call this in the directory with your code
```

Comment:
Remind them that any time they change the `CMakeLists.txt` they'll want to rerun `cmake ..`, but otherwise they can just run `make`